### PR TITLE
feat: [CDS-42953]: Custom Artifact trigger changes

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38263)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38263)

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -258,7 +258,7 @@ pmsGrpcClientConfig:
   target: localhost:12011
   authority: localhost
 
-shouldConfigureWithPMS: ${SHOULD_CONFIGURE_WITH_PMS:-false}
```
Comment (puthrayaharness) [inline 120-ng-manager/config.yml:261]: Please revert

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by vtxorxwitty — APPROVED** [review]
:lgtm:

> Reviewed 3 of 21 files at r1, 77 of 78 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38263#-:-NDm9hEs5UvIPgg5avj7:bnfp4nl)* status: :shipit: complete! ✓

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 3 of 21 files at r1, 77 of 78 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38263)* status: :shipit: complete! ✓

**Review by Jimit09Gandhi — COMMENTED** [review]
> Reviewed 3 of 21 files at r1, 77 of 78 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38263)* status: :shipit: complete! ✓

___
*[`access-control/service/src/main/java/io/harness/accesscontrol/commons/version/VersionInfoResource.java` line 2 at r3](https://reviewable.io/reviews/harness/harness-core/38263#-NDmi4t_9TSFEL-l_zmq:-NDmi4t_9TSFEL-l_zmr:b-epz65v) ([raw file](https://github.com/harness/harness-core/blob/31f05ba0c11ba92bfd2a70c73fc6715e8f3aa763/access-control/service/src/main/java/io/harness/accesscontrol/commons/version/VersionInfoResource.java#L2)):*
> ```Java
> /*
>  * Copyright 2022 Harness Inc. All rights reserved.
> ```

Thanks for adding copyright info but I think it is better to have it as part of another PR and not mix with functional change as it may be difficult to track.

**Review by reetika04 — APPROVED** [review]
> Reviewed 3 of 21 files at r1, 77 of 78 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38263)* status: :shipit: complete! ✓

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38263)* status: :shipit: complete! ✓

**Comment by shivamnegi94** [comment]
trigger ti3, ti4

**Pull request merged**